torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -6686,7 +6686,11 @@ class Bar:
 
             b = Bar()
             gather_objects = [b for _ in range(dist.get_world_size())]
-            with self.assertRaisesRegex(AttributeError, "Can't pickle local object"):
+            if sys.version_info >= (3, 12):
+                error_message = "Can't get local object"
+            else:
+                error_message = "Can't pickle local object"
+            with self.assertRaisesRegex(AttributeError, error_message):
```

Comment:
The change seems pretty innocuous, but I think we shouldn't regex the message at all if its dependent on python. You can just update it to assert it raises `AttributeError`

Reply:
Done.